rl4co/models/zoo/am/decoder.py

```diff
@@ -108,7 +108,7 @@ def __init__(
 
         if pointer is None:
             # MHA with Pointer mechanism (https://arxiv.org/abs/1506.03134)
-            pointer = PointerAttention(
```

Comment:
So we are defaulting to `PointerAttnMoE` here for the base AM class? I think this might be confusing for some who just want the default model (without MoE). I would rather keep the previous version in this case (`PointerAttention` but with MoE kwargs)

Or, I would suggest to do this:
```python
pointer_attn_class = PointerAttention if moe_kwargs is None else PointerAttnMoE
pointer = pointer_attn_class(embed_dim ...)
```

Reply:
The second one looks good! Please see the latest commit.